server/src/test/java/org/elasticsearch/node/NodeTests.java

```diff
@@ -180,7 +181,11 @@ public void testCloseRaceWithTaskExecution() throws Exception {
                 throw new AssertionError("interrupted while waiting", e);
             }
             threadpool.executor(ThreadPool.Names.SEARCH).execute(() -> {
-                while (shouldRun.get());
+                try {
+                    while (shouldRun.get());
+                } catch (RejectedExecutionException e) {
+                    assertThat(e.getMessage(), containsString("[Terminated,"));
+                }
```

Comment:
I think you should put the call to `execute` in the try block, not `shouldRun.get()`, which never throws rejected execution exceptions?

Reply:
thank you! I tried to do this too quickly and this is what I get